datalad_next/iter_collections/gitworktree.py

```diff
@@ -93,6 +93,7 @@ def iter_gitworktree(
     untracked: str | None = 'all',
     link_target: bool = False,
     fp: bool = False,
+    recursive: str = 'repository',
```

Comment:
What is the rationale for using the string `'repository'`? It is not validated, anything apart from `'no'` would lead to identical behavior. So it seems more like a `'no'` and not-`'no'` value. Wouldn't a `bool` fit better? Or are we expecting additional recursion behaviors? In the latter case, we might want to check for known behaviors, i.e. for `'repository'`

Reply:
Yes, there will be subdataset recursion. The corresponding test already has a hint at that.